pandas/core/strings.py

```diff
@@ -1314,21 +1314,65 @@ def str_index(arr, sub, start=0, end=None, side='left'):
 
 def str_pad(arr, width, side='left', fillchar=' '):
     """
-    Pad strings in the Series/Index with an additional character to
-    specified side.
+    Pad strings in the Series/Index up to width.
 
     Parameters
     ----------
     width : int
         Minimum width of resulting string; additional characters will be filled
-        with spaces
+        with character defined in fillchar.
     side : {'left', 'right', 'both'}, default 'left'
-    fillchar : str
-        Additional character for filling, default is whitespace
+        Side from which to fill resulting string.
+    fillchar : str, default ' '
+        Additional character for filling, default is whitespace.
 
     Returns
     -------
-    padded : Series/Index of objects
+    Series or Index of object
+        Returns Series or Index with minimum number of char in object.
+
+    Examples
+    --------
+    >>> s = pd.Series(["caribou", "tiger"])
+    >>> s
+    0    caribou
+    1      tiger
+    dtype: object
+
+    >>> s.str.pad(width=10)
+    0       caribou
+    1         tiger
+    dtype: object
+
+    >>> s.str.pad(width=10, side='right', fillchar='-')
+    0    caribou---
+    1    tiger-----
+    dtype: object
+
+    >>> s.str.pad(width=10, side='both', fillchar='-')
+    0    -caribou--
+    1    --tiger---
+    dtype: object
+
+    >>> s = pd.Series(["caribou", 0, np.NaN])
+    >>> s.str.pad(width=10, fillchar='-')
+    0    ---caribou
+    1           NaN
+    2           NaN
+    dtype: object
+
+    See Also
+    --------
+    Series.str.rjust: Fills the left side of strings with an arbitrary
+        character.
+    Series.str.ljust: Fills the right side of strings with an arbitrary
+        character.
+    Series.str.pad: Fills the specified sides of strings with an arbitrary
+        character.
+    Series.str.center: Fills boths sides of strings with an arbitrary
+        character.
+    Series.str.zfill:  Pad strings in the Series/Index by prepending '0'
+        character.
```

Comment:
`See Also` should be placed before the `Examples`. You have the documentation about docstrings here: https://pandas.pydata.org/pandas-docs/stable/contributing_docstring.html

As `pad` is this same method, I wouldn't include it in the list of related methods.

Did you try rendering the page with `./doc/make.py html --single pandas.Series.str.pad`? I think there is a missing space before the colon that will break the format.

Also, if you run `./scripts/validate_docstrings.py pandas.Series.str.pad` will probably report it.

More than explaining `rjust`... I'd just say it's equivalent to `pad(side='left')`, so users don't need to guess how they are different and which they should use.